src/details/ArborX_DetailsDBSCANCallback.hpp

```diff
@@ -123,7 +122,10 @@ struct DBSCANCallback
   }
 
   KOKKOS_FUNCTION
-  void combine_trees(int self, int neighbor) const
+  void union_1way(int i, int j) const { labels_(i) = representative(j); }
+
+  KOKKOS_FUNCTION
+  void union_2way(int i, int j) const
```

Comment:
Why union "one" and "two" way?  I don't get it.

Reply:
Union-find algorithm essentially merges two (implicit) trees. In general, it does not matter which one is a root, and which one is merged into, and is usually based on which representative is smaller. However, for the boundary points, it matters, as otherwise it may lead to the bridging effect. So the boundary point tree has to be merged into the core point tree, which is what I call `union_1way`, which simply assigns a label.

Reply:
I still don't get the name

Reply:
Given what I described, how would you propose to call them?

Reply:
Maybe be more explicit via `union_boundary_core` or such?

Reply:
Union-find structure is a general concept outside of DBSCAN. I would want to limit its scope to its basic functionality, including the names.

Reply:
I would just not include `union_1way` in `UnionFind` then (and have the behavior in-line as before) and don't change the name of `union_2way`.

Reply:
@masterleinad I need union 1 way in UnionFind, as I do not want to expose labels.

@masterleinad @dalg24 Would you guys feel better if I change the signature to
`union(int i, int j, bool one_way = false);`? It may be slightly less performant, as it will introduce an extra unnecessary `if` check, but should not be a biggie.

Reply:
I think it's fine to have two member functions in that case but they should be self-describing. What you essentially want is a version that assigns that updates the left tree ( or the right tree), isn't it?
 If so, `union_merge_left` and `union_merge_any` (or just `union`) would be fine with me. Alternatively, introducing a template argument `force_merge_left` should not have any runtime overhead and might be nicer by providing only one function.

Reply:
> What you essentially want is a version that assigns that updates the left tree ( or the right tree), isn't it?

Well, similar, but there is no left tree and right tree. There are two trees, one consisting of a single vertex (corresponding to the boundary point), and one other tree. To me, `left` is completely confusing, and does not make sense.

Reply:
OK. Then calling this function has special requisites, i.e. one of them having only a single vertex, in that case I would certainly use two different names, maybe `union_with_single_vertex_tree` or so.